### ISSUE TITLE
Add URL to RequestRejectedException when rejecting requests

### ISSUE THREAD

**Issue by PatrickGotthard** [spring-projects/spring-security#5107]
### Summary

When a request gets rejected by the StrictHttpFirewall the URL will not be logged. So it's very hard to find out which client, frontend or else caused the issue.

### Actual Behavior

When a request gets rejected by the StrictHttpFirewall the error message only states why a request was blocked but not which URL was called.

### Expected Behavior

When a request gets blocked by the StrictHttpFirewall the exception should contain the URL which was called.

### Version

spring-security-web 5.0.3.RELEASE

**Comment by rwinch** [comment]
Thanks for the report. Is there a reason the server cares? I would think that the client would need to change behavior so it would know what request caused the error.

**Comment by PatrickGotthard** [comment]
Hi @rwinch,

my application is a monolith consisting of a REST backend with a few hundred REST endpoints and an ExtJS frontend. Since upgrading to Spring Boot 2 the application logged some RequestRejectedExceptions but I had no idea what caused the problem. 

I the meantime I've installed a custom ServletFilter that handles the RequestRejectedExceptions and logs the URL that caused the exception but I think it would be much more user friendly when the URL will be logged out-of-the-box.

Regards,
Patrick

**Comment by rwinch** [comment]
Thanks for the additional information. I'm going to log this as a duplicate of #5107 which is already asking for an improved way to handle the `RequestRejectedException`

**Comment by PatrickGotthard** [comment]
Did you mean #5007 ? :)

**Comment by rwinch** [comment]
Thanks updated my comment to make it easier to follow along
